pkg/controller/container-runtime-config/helpers.go

```diff
@@ -440,6 +441,12 @@ func updatePolicyJSON(data []byte, internalBlocked, internalAllowed []string, re
 	if err != nil {
 		return nil, fmt.Errorf("error decoding policy json: %w", err)
 	}
+	scopeRegexp := "^[^/]*(/[^:/]*(/[^:/]*(:[^:/]*)?)?)?$"
```

Comment:
This regexp hurts my head...would anything fail here if we just used `strings.Count(s, "/")`?

This regexp could use some direct unit tests too.

Reply:
I agree, `strings.Count` seems far simpler . Using this regexp would really warrant splitting it into a separate function with cases of `?` matching/not matching tested.